spec.md

```diff
@@ -1116,16 +986,24 @@ message NodeServiceCapability {
 }
 ```
 
-### Error Type/Codes
+### Error Handling
 
-Standard error types that Plugins MUST reply with.
-Allows CO to implement smarter error handling and fault resolution.
+All CSI API calls defined in this spec MUST return a [standard gRPC status](https://github.com/grpc/grpc/blob/master/src/proto/grpc/status/status.proto).
+Most gRPC libraries provide helper methods to set and read the status fields.
+
+The status `code` MUST contain a [canonical error code](https://github.com/grpc/grpc-go/blob/master/codes/codes.go).
+The status `message` MUST contain a human readable description of error, if the status `code` is not `OK`.
+This string MAY be surfaced by CO to end users.
+The status `details` MUST return the machine-parsable `Error` protobuf message defined below, if the status `code` is not `OK`.
+This allows CO to implement smarter error handling and fault resolution.
 
 ```protobuf
 message Error {
   // General Error that MAY be returned by any RPC.
   message GeneralError {
```

Comment:
Sounds like we don't need to define the enum for general errors as we'll use the general grpc error code. Here is my suggestion:
1) For each general error (that applies to all RPCs) code from grpc that we'll use, have a text to explain what that means in this section (like what we have for now), and define the corresponding error details struct. We can define a GeneralErrorDetails message that'll be reused by all general errors.
2) For RPC specific errors, we should place the corresponding error explanation close to where the RPC is. We might want to define RPC specific error details type.